core/src/main/java/org/axonframework/queryhandling/QueryUpdateEmitterResolverFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.axonframework.queryhandling;
+
+import org.axonframework.common.Assert;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.axonframework.messaging.unitofwork.CurrentUnitOfWork;
+
+import java.lang.reflect.Executable;
+import java.lang.reflect.Parameter;
+
+/**
+ * Parameter resolver factory that enables injection of {@link QueryUpdateEmitter} to query handlers.
+ * <p>
+ * Injection of correct {@link QueryUpdateEmitter} is achieved using {@link QueryUpdateEmitterResolverFactory#initialize(QueryUpdateEmitter)}.
+ *
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class QueryUpdateEmitterResolverFactory
```

Comment:
Curious: why didn't you have an inner public `ParameterResolver` implementing class like the other `ParameterResolver` implementations in the code base?

Reply:
This class is similar to `ConflictResolution`. Do you still think we should refactor this?

Reply:
No you don't necessarily have to refactor it if you'd ask me. I was just curious why you took this form rather than the others :)